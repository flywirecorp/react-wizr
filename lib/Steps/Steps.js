import React from 'react';
import PropTypes from 'prop-types';

const Steps = ({ children, ...rest }, { activeStepIndex }) => {
  const currentStep = React.Children.toArray(children)[activeStepIndex];

  return <div {...rest}>{currentStep}</div>;
};

Steps.contextTypes = {
  activeStepIndex: PropTypes.number.isRequired
};

export default Steps;
