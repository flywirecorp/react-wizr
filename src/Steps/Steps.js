import React from 'react';
import PropTypes from 'prop-types';

const Steps = ({ children }, { activeStepIndex }) => {
  const currentStep = React.Children.toArray(children)[activeStepIndex];

  return currentStep;
};

Steps.defaultProps = {
  isSteps: true
};

Steps.contextTypes = {
  activeStepIndex: PropTypes.number.isRequired
};

export default Steps;
