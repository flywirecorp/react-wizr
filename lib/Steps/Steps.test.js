import React from 'react';
import { shallow } from 'enzyme';
import Steps from './Steps';

describe('Steps', () => {
  it('renders the correct child', () => {
    const Step = () => <div />;
    const context = { activeStepIndex: 1 };
    const wrapper = shallow(
      <Steps>
        <Step />
        <Step />
      </Steps>,
      { context }
    );

    expect(wrapper).toMatchSnapshot();
  });
});
