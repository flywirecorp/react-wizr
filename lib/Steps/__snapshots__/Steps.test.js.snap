// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps renders the correct child 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Steps>
    <Step />
    <Step />
  </Steps>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Step />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": ".1",
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Step />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "activeStepIndex": 1,
    },
  },
}
`;
